CMakeLists.txt

```diff
@@ -19,6 +19,17 @@
 cmake_minimum_required(VERSION 3.6)
 project(laminar)
 
+if (${CMAKE_SYSTEM_NAME} STREQUAL "FreeBSD")
+    if (NOT DEFINED LINKER_EMULATION_FLAGS)
+        if (${CMAKE_SYSTEM_PROCESSOR} STREQUAL "amd64")
+            set(LINKER_EMULATION_FLAGS "-melf_x86_64")
+        else()
+            message(FATAL_ERROR "Unsupported 'uname -m' translation ('${CMAKE_SYSTEM_PROCESSOR}') to emulation flag")
+        endif()
```

Comment:
Passing `-m` doesn't hurt on Linux, so it would be nice to avoid making this conditional on FreeBSD. On my desktop `CMAKE_SYSTEM_PROCESSOR` returns `x86_64` though. And laminar is used on 32 and 64 bit ARM systems too.

Reply:
(a) I will update the condition to detect LD.LLD linker. Gnu Linker uses a default builtin emulation type, so it does not require  the "-m" flag. LD.LLD is not able to do that. I don't want to enforce the "-m" flag when I don't have to.
(b) There are no canonical values of "CMAKE_SYSTEM_PROCESSOR" across different OSes.

Reply:
I'm not able to detect the linker, so I'm using an extra variable LLVM_LINKER_IS_LLD, which is enabled for FreeBSD build.